go/oasis-test-runner/cmd/root.go

```diff
@@ -73,17 +99,194 @@ func Execute() {
+			psCloned[k] = v
+		}
+		return []map[string]string{psCloned}
+	}
+
+	rps := []map[string]string{}
+
+	// Take first element from cloned zp and do recursion deterministically.
+	var zpKeys []string
+	for k := range zp {
+		zpKeys = append(zpKeys, k)
+	}
+	sort.Strings(zpKeys)
+
+	zpCloned := map[string][]string{}
+	for _, k := range zpKeys[1:] {
+		zpCloned[k] = zp[k]
+	}
+	for _, v := range zp[zpKeys[0]] {
+		ps[zpKeys[0]] = v
+		rps = append(rps, computeParamSets(zpCloned, ps)...)
+	}
+
+	return rps
+}
+
+// RegisterDefaultScenarios registers all available default scenarios and returns any errors encountered.
+func RegisterDefaultScenarios() (ers []error) {
+	var errors []error
+	// Basic test.
+	errors = append(errors, Register(e2e.Basic))
+	errors = append(errors, Register(e2e.BasicEncryption))
+	// Byzantine executor node.
+	errors = append(errors, Register(e2e.ByzantineExecutorHonest))
+	errors = append(errors, Register(e2e.ByzantineExecutorWrong))
+	errors = append(errors, Register(e2e.ByzantineExecutorStraggler))
+	// Byzantine merge node.
+	errors = append(errors, Register(e2e.ByzantineMergeHonest))
+	errors = append(errors, Register(e2e.ByzantineMergeWrong))
+	errors = append(errors, Register(e2e.ByzantineMergeStraggler))
+	// Storage sync test.
+	errors = append(errors, Register(e2e.StorageSync))
+	// Sentry test.
+	errors = append(errors, Register(e2e.Sentry))
+	errors = append(errors, Register(e2e.SentryEncryption))
+	// Keymanager restart test.
+	errors = append(errors, Register(e2e.KeymanagerRestart))
+	// Dump/restore test.
+	errors = append(errors, Register(e2e.DumpRestore))
+	// Halt test.
+	errors = append(errors, Register(e2e.HaltRestore))
+	// Multiple runtimes test.
+	errors = append(errors, Register(e2e.MultipleRuntimes))
+	// Registry CLI test.
+	errors = append(errors, Register(e2e.RegistryCLI))
+	// Stake CLI test.
+	errors = append(errors, Register(e2e.StakeCLI))
+	// Node shutdown test.
+	errors = append(errors, Register(e2e.NodeShutdown))
+	// Gas fees tests.
+	errors = append(errors, Register(e2e.GasFeesStaking))
+	errors = append(errors, Register(e2e.GasFeesStakingDumpRestore))
+	errors = append(errors, Register(e2e.GasFeesRuntimes))
+	// Identity CLI test.
+	errors = append(errors, Register(e2e.IdentityCLI))
+	// Runtime prune test.
+	errors = append(errors, Register(e2e.RuntimePrune))
+	// Runtime dynamic registration test.
+	errors = append(errors, Register(e2e.RuntimeDynamic))
+	// Transaction source test.
+	errors = append(errors, Register(e2e.TxSourceMultiShort))
+	errors = append(errors, RegisterNondefault(e2e.TxSourceMulti))
+	// Node upgrade tests.
+	errors = append(errors, Register(e2e.NodeUpgrade))
+	errors = append(errors, Register(e2e.NodeUpgradeCancel))
+	// Debonding entries from genesis test.
+	errors = append(errors, Register(e2e.Debond))
+
+	// Register the remote signer test cases.
+	errors = append(errors, Register(remotesigner.Basic))
+
+	// Filter out nil errors.
+	for _, e := range errors {
```

Comment:
You could avoid the filtering if you did:

```go
	if err := Register(e2e.Basic); err != nil {
	  errors = append(errors, err)
	}
```
or probably even better just return on the first encountered error?

Reply:
ok